docs/source/api-main-setup.rst

```diff
@@ -0,0 +1,167 @@
+
+Every command now should start with postgres=#
+
+In the postgres world, run the following commands:
+
+``#\password postgres``
+
+You should be prompted to enter a new password. Don’t type anything, just hit enter twice. This should clear the password.
+
+postgres=# ``create database lookit``
+
+postgres=# ``grant all privileges on database lookit to postgres``
+
+If at this point you still do not have access to the lookit database, run the following commands:
+
+``sudo vi /etc/postgresql/10/main/pg_hba.conf``
+
+``pg_ctl reload``
+
+A long document with # on the leftmost side of almost every line should open up. Scroll to the bottom. There will be a few lines that don’t start with #. They might be a different color and will start with either local or host. The last word in each of those lines should be trust. If it is not, switch into editing mode ( hit esc then type i and hit enter ) and change them to say trust. Then, save the file ( hit esc and then type :x before hitting enter ). You should now have access.
+
+
+
+
+
+Install RabbitMQ
+----------------
+
+**For Linux Users**
+
+First, run the following command:
+
+``sudo apt install rabbitmq-server``
+
+Now that rabbitmq server is installed, create an admin user with the following commands:
+
+``sudo rabbitmqctl add_user admin password``
+
+``sudo rabbitmqctl set_user_tags admin administrator``
+
+``sudo rabbitmqctl set_permissions -p / admin ".*" ".*" ".*"``
+
+Make sure that the server is up and running:
+
+``sudo systemctl stop rabbitmq-server.service``
+
+``sudo systemctl start rabbitmq-server.service``
+
+``sudo systemctl enable rabbitmq-server.service``
+
+If you are having problems creating a user or getting the server running, try the following commands:
+
+``sudo rabbitmq-plugins enable rabbitmq_management``
+
+``sudo chown -R rabbitmq:rabbitmq /var/lib/rabbitmq/``
+
+``sudo rabbitmqadmin declare queue --vhost=/ name=email``
+
+``sudo rabbitmqadmin declare queue --vhost=/ name=builds``
+
+``sudo rabbitmqadmin list queues``
+
+When you run the last command, you should see something like this: ::
+
+    +--------+----------+
+    | name  | messages  |
+    +--------+----------+
+    | builds|     0     |
+    | email |     0	|
+    +--------+----------+
+
+
+Install Ngrok
+-------------
+
+**For Linux Users**
+
+To install, run this command:
+
+``sudo snap install ngrok``
+
+To connect to your local host run this command:
+
+``ngrok http “[https://localhost:8000](https://localhost:8000)"``
+
+
+
+Setup
+~~~~~
+
+
+
+
+
+
+
+
+How Do These Programs Work Together?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following diagrams illustrate how different parts of the API interact with each other.
+
+.. image:: _static/img/RabbitMQ.png
+
+Every time the user makes a request, it is sent through microWSGI. If it is a request that will take more than a few
+seconds, it is sent off to RabbitMQ which will delegate the tasks one by one to celery. If it is short enough, HTTP will
```

Comment:
Follow the semantic structure of the first clause of your sentence, and don't just avoid passive voice - figure out what is doing what! That way, when you DO use passive voice, the reader takes the hint that you are talking about *process* or *structure* rather than *action*.

`For certain requests that take more than a few seconds (e.g., building study dependencies, sending mail, etc.), the WSGI handler defers the work by kicking off a new celery task. The payload for this task is mediated by the RabbitMQ service.`